go/libraries/doltcore/sqle/binlogreplication/binlog_replication_test.go

```diff
@@ -0,0 +1,792 @@
+		"--datadir="+dataDir,
+		"--default-authentication-plugin=mysql_native_password")
+	output, err = chmodCmd.CombinedOutput()
+	if err != nil {
+		return -1, nil, fmt.Errorf("unable to execute command %v: %v – %v", cmd.String(), err.Error(), string(output))
+	}
+
+	cmd = exec.Command("mysqld",
+		"--no-defaults",
+		"--user="+username,
+		"--datadir="+dataDir,
+		"--default-authentication-plugin=mysql_native_password",
+		"--gtid-mode=ON",
+		"--enforce-gtid-consistency=ON",
+		fmt.Sprintf("--port=%v", mySqlPort),
+		"--server-id=11223344",
+		fmt.Sprintf("--socket=mysql-%v.sock", mySqlPort),
+		"--binlog-checksum=NONE",
+		"--general_log_file="+dir+"general_log",
+		"--log-bin="+dir+"log_bin",
+		"--slow_query_log_file="+dir+"slow_query_log",
+		"--log-error="+dir+"log_error",
+		fmt.Sprintf("--pid-file="+dir+"pid-%v.pid", mySqlPort))
+
+	mysqlLogFilePath = filepath.Join(dir, fmt.Sprintf("mysql-%d.out.log", time.Now().Unix()))
+	mysqlLogFile, err = os.Create(mysqlLogFilePath)
+	if err != nil {
+		return -1, nil, err
+	}
+	fmt.Printf("MySQL server logs at: %s \n", mysqlLogFilePath)
+	cmd.Stdout = mysqlLogFile
+	cmd.Stderr = mysqlLogFile
+	err = cmd.Start()
+	if err != nil {
+		return -1, nil, fmt.Errorf("unable to start process %q: %v", cmd.String(), err.Error())
+	}
+
+	dsn := fmt.Sprintf("root@tcp(127.0.0.1:%v)/", mySqlPort)
+	primaryDatabase = sqlx.MustOpen("mysql", dsn)
+
+	err = waitForSqlServerToStart(primaryDatabase)
+	if err != nil {
+		return -1, nil, err
+	}
+
+	// Create the initial database on the MySQL server
+	primaryDatabase.MustExec("create database db01;")
+
+	dsn = fmt.Sprintf("root@tcp(127.0.0.1:%v)/db01", mySqlPort)
+	primaryDatabase = sqlx.MustOpen("mysql", dsn)
+
+	os.Chdir(originalCwd)
+
+	fmt.Printf("MySQL server started on port %v \n", mySqlPort)
+
+	return mySqlPort, cmd.Process, nil
+}
+
+func initializeDevDoltBuild(dir string, goDirPath string) string {
+	// If we're not in a CI environment, don't worry about building a dev build
+	if os.Getenv("CI") != "true" {
+		return ""
+	}
+
+	basedir := filepath.Dir(filepath.Dir(dir))
+	fullpath := filepath.Join(basedir, fmt.Sprintf("devDolt-%d", os.Getpid()))
+
+	_, err := os.Stat(fullpath)
+	if err == nil {
+		return fullpath
+	}
+
+	fmt.Printf("building dolt dev build at: %s \n", fullpath)
+	cmd := exec.Command("go", "build", "-o", fullpath, "./cmd/dolt")
+	cmd.Dir = goDirPath
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		panic("unable to build dolt for binlog integration tests: " + err.Error() + "\nFull output: " + string(output) + "\n")
+	}
+	return fullpath
+}
+
+func startDoltSqlServer(dir string) (int, *os.Process, error) {
+	dir = filepath.Join(dir, "dolt")
+	err := os.MkdirAll(dir, 0777)
+	if err != nil {
+		return -1, nil, err
+	}
+
+	doltPort = findFreePort()
+	fmt.Printf("Starting Dolt sql-server on port: %d, with data dir %s\n", doltPort, dir)
+
+	// take the CWD and move up four directories to find the go directory
+	if originalWorkingDir == "" {
```

Comment:
Yeah... this is a little hacky. The test code needs to find-up the dir tree until it finds the base `go` directory so that it can run `go run ./cmd/dolt` (in CI we actually `go build ./cmd/dolt` to save time/cpu from rebuilding over and over). Since the tests are invoked in this dir, this seems to work for running in GoLand, on the commandline, or in CI, but it could definitely be cleaned up more.